ut/alarm_tables_testing.cpp

```diff
@@ -0,0 +1,321 @@
+/**
+ * Project Clearwater - IMS in the cloud.
+ * Copyright (C) 2015  Metaswitch Networks Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version, along with the "Special Exception" for use of
+ * the program along with SSL, set forth below. This program is distributed
+ * in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+ * A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details. You should have received a copy of the GNU General Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ *
+ * The author can be reached by email at clearwater@metaswitch.com or by
+ * post at Metaswitch Networks Ltd, 100 Church St, Enfield EN2 6BQ, UK
+ *
+ * Special Exception
+ * Metaswitch Networks Ltd  grants you permission to copy, modify,
+ * propagate, and distribute a work formed by combining OpenSSL with The
+ * Software, or a work derivative of such a combination, even if such
+ * copying, modification, propagation, or distribution would otherwise
+ * violate the terms of the GPL. You must comply with the GPL in all
+ * respects for all of the code used other than OpenSSL.
+ * "OpenSSL" means OpenSSL toolkit software distributed by the OpenSSL
+ * Project and licensed under the OpenSSL Licenses, or a work based on such
+ * software and licensed under the OpenSSL Licenses.
+ * "OpenSSL Licenses" means the OpenSSL License and Original SSLeay License
+ * under which the OpenSSL Project distributes the OpenSSL toolkit software,
+ * as those licenses appear in the file LICENSE-OPENSSL.
+ */
+
+#include "test_snmp.h"
+#include "alarm_model_table.hpp"
+#include "itu_alarm_table.hpp"
+#include "alarm_active_table.hpp"
+#include "utils.h"
+#include "stdlib.h"
+#include <algorithm>
+#include <string>
+
+std::string alarmModelTable_entry = "1.3.6.1.2.1.118.1.1.2.1";
+std::string ituAlarmTable_entry = "1.3.6.1.2.1.121.1.1.1.1";
+std::string alarmActiveTable_table = "1.3.6.1.2.1.118.1.2.2";
+
+char buf[1024];
+
+TEST_F(SNMPTest, ModelTable)
+{ 
+  // Create a custom alarm definition
+  AlarmDef::SeverityDetails cleared(AlarmDef::CLEARED, "Test alarm cleared description", "Test alarm cleared details");
```

Comment:
I would be tempted to bring all this alarm creation code into a common SetUp function, and make defN_raised, defN_cleared, etc. into member variables. (You'd need to subclass SNMPTest to do this.) This means that it's a bit clearer what your code is actually testing - you don't start off with several lines of boilerplate just to create these.